Sources/Helpers/OAWikiImageHelper.mm

```diff
@@ -46,18 +53,143 @@ - (void)sendNearbyWikiImagesRequest:(OARowInfo *)nearbyImagesRowInfo targetObj:(
+                NSString *lang = [[prefix substringFromIndex:[prefix lastIndexOf:@"/"] + 1] stringByReplacingOccurrencesOfString:@"/" withString:@""];
+                NSString *title = [wikiTitle substringFromIndex:urlInd + @".wikipedia.org/wiki/".length];
+                wikiTitle = [NSString stringWithFormat:@"%@:%@", lang, title];
+            }
+
+            url = url.length == 0
+                ? [NSString stringWithFormat:@"%@%@%@", OSMAND_API_ENDPOINT, @"wiki=", wikiTitle]
+                : [url stringByAppendingFormat:@"&%@%@", @"wiki=", wikiTitle];
+        }
+
+        if (USE_OSMAND_WIKI_API)
+        {
+            if (url.length > 0)
+            {
+                url = [url stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]];
+                [self addOsmandAPIImageList:url cards:cards];
+            }
+            else if (_addOtherImagesFunction)
+            {
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    _addOtherImagesFunction(cards);
+                });
+            }
+        }
+        else
+        {
+            if (wikidataId.length == 0 && wikiCategory.length == 0)
+            {
+                if (_addOtherImagesFunction)
+                {
+                    dispatch_async(dispatch_get_main_queue(), ^{
+                        _addOtherImagesFunction(cards);
+                    });
+                }
+            }
+            else
+            {
+                _cardTypesReady = [NSMutableArray array];
+                _foundImages = [NSMutableArray array];
+                
+                if (wikidataId.length == 0)
+                    [_cardTypesReady addObject:@(EOAWikiImageTypeWikidata)];
+                else
+                    [self addWikidataImageCards:wikidataId cards:cards];
+                
+                if (wikiCategory.length == 0)
+                    [_cardTypesReady addObject:@(EOAWikiImageTypeWikimedia)];
+                else
+                    [self addWikimediaCardsFromCategory:wikiCategory cards:cards depth:0 prepared:YES];
+            }
+        }
+    }
+    else if (_addOtherImagesFunction)
+    {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            _addOtherImagesFunction(cards);
+        });
     }
-    _wikidataCardsReady = NO;
-    _wikimediaCardsReady = NO;
-    [self addWikimediaCards:wikimediaTagContent cards:cards];
-    [self addWikidataCards:wikidataTagContent cards:cards];
+}
+
+- (void)addOsmandAPIImageList:(NSString *)url cards:(NSMutableArray<OAAbstractCard *> *)cards
+{
+    NSURL *urlObj = [[NSURL alloc] initWithString:url];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]];
+    [[session dataTaskWithURL:urlObj completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+        if (((NSHTTPURLResponse *)response).statusCode == 200)
```

Comment:
when working with network requests, it is advisable to add logs in all cases of incorrect response or parsing